src/PrestaShopBundle/Controller/Admin/Improve/Design/ImageSettingsController.php

```diff
@@ -0,0 +1,374 @@
+        /** @var ResponseBuilder $responseBuilder */
+        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
+
+        return $responseBuilder->buildSearchResponse(
+            $this->get('prestashop.core.grid.definition.factory.image_type'),
+            $request,
+            ImageTypeGridDefinitionFactory::GRID_ID,
+            'admin_image_settings_index'
+        );
+    }
+
+    /**
+     * Regenerate image thumbnails.
+     *
+     * @AdminSecurity(
+     *     "is_granted(['update'], request.get('_legacy_controller'))",
+     *     redirectRoute="admin_image_settings_index"
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function regenerateThumbnailsAction(Request $request): RedirectResponse
+    {
+        $form = $this->createForm(ThumbnailRegenerationType::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+            $errors = $this->get('prestashop.adapter.image_type.thumbnail_regenerator')->regenerateThumbnails($data);
+
+            if (!empty($errors)) {
+                $this->flashErrors($errors);
+            }
+        }
+
+        return $this->redirectToRoute('admin_image_settings_index');
+    }
+
+    /**
+     * Moves images from old file system to new.
+     *
+     * @AdminSecurity(
+     *     "is_granted(['update'], request.get('_legacy_controller'))",
+     *     redirectRoute="admin_image_settings_index"
+     * )
+     *
+     * @return RedirectResponse
+     */
+    public function moveImagesAction(): RedirectResponse
+    {
+        $errors = $this->get('prestashop.adapter.image_settings.helper')->moveImagesToNewFileSystem();
+        if (count($errors)) {
+            $this->flashErrors($errors);
+        }
+
+        return $this->redirectToRoute('admin_image_settings_index');
+    }
+
+    /**
+     * Show image type creation form page and handle its submit.
+     *
+     * @AdminSecurity("is_granted('create', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request): Response
+    {
+        $imageTypeFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.image_type_form_handler');
+        $imageTypeFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.image_type_form_builder');
+
+        $imageTypeForm = $imageTypeFormBuilder->getForm();
+        $imageTypeForm->handleRequest($request);
+
+        try {
+            $result = $imageTypeFormHandler->handle($imageTypeForm);
+
+            if (null !== $result->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful creation', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_image_settings_index');
+            }
+        } catch (Exception $exception) {
```

Comment:
Maybe too generci :thinking:

Reply:
It is meant to catch everything and relay it to an error flash message. That's why it is so generic.